src/viam/sdk/components/sensor/client.hpp

```diff
@@ -0,0 +1,52 @@
+/// @file components/sensor/client.hpp
+///
+/// @brief Implements a gRPC client for the `Sensor` component.
+#pragma once
+
+#include <grpcpp/channel.h>
+
+#include <viam/api/component/sensor/v1/sensor.grpc.pb.h>
+
+#include <viam/sdk/components/sensor/sensor.hpp>
+#include <viam/sdk/components/sensor/server.hpp>
+#include <viam/sdk/config/resource.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @class SensorClient
+/// @brief gRPC client implementation of a `Sensor` component.
+/// @ingroup Sensor
+class SensorClient : public Sensor {
+   public:
+    SensorClient(std::string name, std::shared_ptr<grpc::Channel> channel);
+    AttributeMap get_readings(const AttributeMap& extra) override;
+    AttributeMap do_command(const AttributeMap& command) override;
+    std::vector<GeometryConfig> get_geometries(const AttributeMap& extra) override;
+
+    typedef viam::component::sensor::v1::SensorService::StubInterface Stub;
+
+   private:
+    // template to wrap a stub
+    template <typename Request, typename Response, typename Cls>
+    void stub_wrapper(Cls* self,
```

Comment:
I copied the `stub_wrapper` and `server_wrapper` templated methods from @abe-winter's movement sensor wrapper implementation. I stopped short of moving them to some central location and employing them for every resource.

Reply:
Sounds good; I'll wait for your feedback before proceeding, then. Thanks for thinking about it, would be great to standardize these in an idiomatic way now before we implement more wrappers (although only the power sensor is in my TODO for now).

Reply:
SGTM!